graphstorm-processing/tests/test_dist_heterogenous_loader.py

```diff
@@ -1033,3 +1033,158 @@ def test_edge_custom_label(spark, dghl_loader: DistHeterogeneousGraphLoader, tmp
+    val_total_ones = val_mask_df.agg(F.sum("custom_split_val_mask")).collect()[0][0]
+    test_total_ones = test_mask_df.agg(F.sum("custom_split_test_mask")).collect()[0][0]
+    assert train_total_ones == 999
+    assert val_total_ones == 99
+    assert test_total_ones == 99
+
+    # Check the order of the train_mask_df
+    train_mask_df = train_mask_df.withColumn("order_check_id", F.monotonically_increasing_id())
+    val_mask_df = val_mask_df.withColumn("order_check_id", F.monotonically_increasing_id())
+    test_mask_df = test_mask_df.withColumn("order_check_id", F.monotonically_increasing_id())
+    train_mask_df = train_mask_df.filter((F.col("order_check_id") < 999)).drop("order_check_id")
+    val_mask_df = val_mask_df.filter(
+        (F.col("order_check_id") >= 1000) & (F.col("order_check_id") < 1099)
+    ).drop("order_check_id")
+    test_mask_df = test_mask_df.filter(
+        (F.col("order_check_id") >= 1100) & (F.col("order_check_id") < 1199)
+    ).drop("order_check_id")
+
+    train_unique_rows = train_mask_df.distinct().collect()
+    assert len(train_unique_rows) == 1 and all(value == 1 for value in train_unique_rows[0])
+    val_unique_rows = val_mask_df.distinct().collect()
+    assert len(val_unique_rows) == 1 and all(value == 1 for value in val_unique_rows[0])
+    test_unique_rows = test_mask_df.distinct().collect()
+    assert len(test_unique_rows) == 1 and all(value == 1 for value in test_unique_rows[0])
+
+
+def test_edge_custom_label_multitask(spark, dghl_loader: DistHeterogeneousGraphLoader, tmp_path):
+    """Test using custom label splits for edges"""
+    data = [(i, j) for i in range(1, 4) for j in range(0, 1000)]
+    # Create DataFrame
+    edges_df = spark.createDataFrame(data, ["src_str_id", "dst_str_id"])
+
+    train_df = spark.createDataFrame(
+        [(i, j) for i in range(1, 2) for j in range(0, 10)],
+        ["mask_src_id", "mask_dst_id"],
+    )
+    val_df = spark.createDataFrame(
+        [(i, j) for i in range(2, 3) for j in range(0, 10)],
+        ["mask_src_id", "mask_dst_id"],
+    )
+    test_df = spark.createDataFrame(
+        [(i, j) for i in range(3, 4) for j in range(0, 10)],
+        ["mask_src_id", "mask_dst_id"],
+    )
+
+    class_mask_names = [
+        f"custom_split_train_mask",
+        f"custom_split_val_mask",
+        f"custom_split_test_mask",
+    ]
+    train_df.repartition(1).write.parquet(f"{tmp_path}/train.parquet")
+    val_df.repartition(1).write.parquet(f"{tmp_path}/val.parquet")
+    test_df.repartition(1).write.parquet(f"{tmp_path}/test.parquet")
+    config_dict = {
+        "column": "",
+        "type": "link_prediction",
+        "custom_split_filenames": {
+            "train": f"{tmp_path}/train.parquet",
+            "valid": f"{tmp_path}/val.parquet",
+            "test": f"{tmp_path}/test.parquet",
+            "column": ["mask_src_id", "mask_dst_id"],
+        },
+        "mask_field_names": class_mask_names,
+    }
```

Comment:
This config only includes a single label right? Can we use a multi-label config instead?

Reply:
I see the point. I have added the multiple config for both node and edge config.